### ISSUE TITLE
Check warnings with clang

### ISSUE THREAD

**Issue by andrew4328** [Lambda-Mountain-Compiler-Backend/lambda-mountain#947]
```
clang BOOTSTRAP/cli.c -o tmp
BOOTSTRAP/cli.c:30142:10: warning: passing 'unsigned int *' to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
}):(wait(((unsigned int*)(NULL)))));
         ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/sys/wait.h:77:27: note: passing argument to parameter '__stat_loc' here
extern __pid_t wait (int *__stat_loc);
                          ^
BOOTSTRAP/cli.c:31391:31: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
({while((uuid_000000000006c6a4==(1))){((void)(({({uuid_000000000006c6a5=uuid_000000000006c694;({});})
         ~~~~~~~~~~~~~~~~~~~~~^~~~~
BOOTSTRAP/cli.c:31391:31: note: remove extraneous parentheses around the comparison to silence this warning
({while((uuid_000000000006c6a4==(1))){((void)(({({uuid_000000000006c6a5=uuid_000000000006c694;({});})
        ~                     ^    ~
BOOTSTRAP/cli.c:31391:31: note: use '=' to turn this equality comparison into an assignment
({while((uuid_000000000006c6a4==(1))){((void)(({({uuid_000000000006c6a5=uuid_000000000006c694;({});})
                              ^~
                              =
2 warnings generated.
```

**Comment by andrew4328** [comment]
clang is a bit more pedantic than gcc. The parentheses thing is 100% just a lint, not warning.

**Comment by andrew4328** [comment]
The wait call is a good warning though. This wouldn't pass the LM typechecker, so it shouldn't pass C without a warning at least.
